vision/emergent_object/emergent_object_processing.py

```diff
@@ -0,0 +1,161 @@
+        The Emergent Object will be placed on the ground and (presumably) not standing up.
+        This means that the area of its bounding box should, in general, be larger than
+        that of the judges.
+
+    AI Confidence Level: 5% of final score:
+        Higher confidence levels should be favored
+
+    Parameters
+    ----------
+    humanoids: list[BoundingBox]
+        A list of all candidate emergent objects. Required attributes:
+            latitude, longitude, bounding_box_area, image_path, confidence
+    odlcs: ODLC_Dict
+        A dictionary containing the locations of all discovered ODLCs
+
+    Returns
+    -------
+    emergent_object: BoundingBox
+        The candidate that has scored the highest, and so is probably the actual emergent
+        object
+    """
+
+    # Weights for each metric
+    category_weights: NDArray[Shape[4], Float64] = np.array([0.5, 0.25, 0.20, 0.05])
+
+    evaluations_list: list[tuple[float, float, float, float]] = []
+
+    subject_humanoid: BoundingBox
+    for subject_humanoid in humanoids:
+        # Get the distance to the closest standard object
+        min_odlc_distance: float = float("inf")
+        odlc: Location
+        for odlc in odlcs.values():
+            odlc_distance: float = geopy.distance.geodesic(
+                (odlc["latitude"], odlc["longitude"]),
+                (subject_humanoid.attributes["latitude"], subject_humanoid.attributes["longitude"]),
+            ).feet
+
+            min_odlc_distance = min(odlc_distance, min_odlc_distance)
+
+        # Get the distance to the closest emergent object that's not in the same image
+        min_humanoid_distance: float = float("inf")
+        humanoid: BoundingBox
+        for humanoid in humanoids:
+            # Don't check candidates that are in the same image
+            if subject_humanoid.attributes["image_path"] != humanoid.attributes["image_path"]:
+                humanoid_distance: float = geopy.distance.geodesic(
+                    (humanoid.attributes["latitude"], humanoid.attributes["longitude"]),
+                    (
+                        subject_humanoid.attributes["latitude"],
+                        subject_humanoid.attributes["longitude"],
+                    ),
+                ).feet
+
+                min_humanoid_distance = min(humanoid_distance, min_humanoid_distance)
+
+        # invert the humanoid distance to prioritize closest humanoids
+        min_humanoid_distance *= -1
+
+        evaluation: tuple[float, float, float, float] = (
+            min_odlc_distance,
+            min_humanoid_distance,
+            subject_humanoid.attributes["bounding_box_area"],
+            subject_humanoid.attributes["confidence"],
+        )
+
+        evaluations_list.append(evaluation)
+
+    evaluation_array: NDArray[Shape["*, 4"], Float64] = np.array(evaluations_list)
+
+    # Get the minimums of each category
+    evaluation_mins: NDArray[Shape["4"], Float64] = np.amin(evaluation_array, axis=0)
```

Comment:
This function is doing a lot. Would be better to split into two separate functions: one for calculating the evaluations and one for choosing the emergent object/further calculations on the evaluations.